src/Raven.Client/Documents/Operations/Backups/BackupConfiguration.cs

```diff
@@ -0,0 +1,127 @@
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.Backups
+{
+    public class BackupConfiguration : IDynamicJson
+    {
+        public string Name { get; set; }
+        public string DatabaseName { get; set; }
+        public BackupType BackupType { get; set; }
+        public SnapshotSettings SnapshotSettings { get; set; }
+        public BackupEncryptionSettings BackupEncryptionSettings { get; set; }
+        public LocalSettings LocalSettings { get; set; }
+        public S3Settings S3Settings { get; set; }
+        public GlacierSettings GlacierSettings { get; set; }
+        public AzureSettings AzureSettings { get; set; }
+        public FtpSettings FtpSettings { get; set; }
+        public GoogleCloudSettings GoogleCloudSettings { get; set; }
+
+        public string GetDefaultTaskName()
```

Comment:
internal

Reply:
moved to `PeriodicBackupConfiguration` because it implements IDatabaseTask